### PR TITLE
TLRC 9863

### PR DESCRIPTION
**Pull request by ibai-gonzalez**

### CODE REVIEW

File: src/Sfa.Tl.ResultsAndCertification.Common.Services/Cache/InMemoryCacheService.cs
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Common.Services/Cache/InMemoryCacheService.cs:1]: I fixed and refactored this class. It was making the system fail in some screens

File: src/Sfa.Tl.ResultsAndCertification.Functions/CoreRommExtract.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            var today = DateTime.UtcNow.Date;
+            bool shouldFunctionRunToday = _configuration.CoreRommValidDateRanges.Any(r => r.Contains(today));
+
+            if (!shouldFunctionRunToday)
+            {
+                await Task.CompletedTask;
+                return;
+            }
+
+            var functionLogDetails = CommonHelper.CreateFunctionLogRequest(context.FunctionName, FunctionType.AnalystCoreResultExtract);
```
Comment (mustafajawad786) [inline src/Sfa.Tl.ResultsAndCertification.Functions/CoreRommExtract.cs:46]: Please can we create new function type as this is different from AnalystCoreResultExtract.

File: src/Sfa.Tl.ResultsAndCertification.Application/Mappers/CoreRommExtractMapper.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+               .ForMember(d => d.RommGrade, opts => opts.MapFrom(s => GetRommGrade(s)))
+               .ForMember(d => d.AppealOpenedTimeStamp, opts => opts.MapFrom(s => GetAppealOpenedTimeStamp(s)))
+               .ForMember(d => d.AppealGrade, opts => opts.MapFrom(s => GetAppealGrade(s)));
+        }
+
+        private string GetCurrentCoreGrade(TqRegistrationPathway tqRegistrationPathway)
+        {
+            var coreGradeResults = tqRegistrationPathway.TqPathwayAssessments
+                                        .SelectMany(p => p.TqPathwayResults)
+                                        .Where(p => !p.PrsStatus.HasValue);
```
[the remark this answers is not in the document]
Comment (mustafajawad786) [reply src/Sfa.Tl.ResultsAndCertification.Application/Mappers/CoreRommExtractMapper.cs:31]: The report prints the old results as well therefore we don't require EndDate==null check here.

File: src/Sfa.Tl.ResultsAndCertification.Functions/CoreRommExtract.cs (hunk 1 above)
Comment (ibai-gonzalez) [reply src/Sfa.Tl.ResultsAndCertification.Functions/CoreRommExtract.cs]: Corrected now

**Review by mustafajawad786 — APPROVED** [review]

**Pull request merged**